my/core/discovery_pure.py

```diff
@@ -114,10 +114,31 @@ def _extract_requirements(a: ast.Module) -> Requires:
 
 # todo should probably be more defensive..
 def all_modules() -> Iterable[HPIModule]:
+    for my_root in _iter_my_roots():
+        yield from _modules_under_root(my_root)
+
+
+def _iter_my_roots() -> Iterable[Path]:
+    import my  # doesn't import any code, because of namespace package
+
+    default_root = Path(__file__).absolute().parent.parent
+
+    try:
+        paths: List[str] = list(my.__path__._path)  # type: ignore[attr-defined]
```

Comment:
hm, I thought one could just use `list(pkg.__path__)`? What's _path for?

Reply:
Does seem to work the same, yeah